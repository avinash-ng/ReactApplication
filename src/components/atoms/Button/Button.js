import React from 'react'
import PropTypes from 'prop-types'
const Button = (props ) => {
    return (
        <button className="a_button" className={props.className} onClick={props.onClicked}>{props.buttonName}</button>
    )
}

Button.propTypes = {
    buttonName: PropTypes.string.isRequired
}

export default Button
