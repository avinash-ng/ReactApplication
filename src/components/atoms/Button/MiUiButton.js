import React from 'react'
import { Button } from "@material-ui/core"
const MiUiButton = props => {
    return (
        <Button className={props.className} style={props.style} variant={props.variant} onClick={props.onClick}>{props.buttonName}</Button> 
    )
}
export default MiUiButton