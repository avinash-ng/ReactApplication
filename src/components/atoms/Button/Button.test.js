import Button from './Button'
import expect from 'expect'
import sinon from 'sinon';
import { shallow, configure } from 'enzyme'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
// import renderer from 'react-test-renderer'
import { render } from '@testing-library/react';

configure({ adapter: new Adapter() });
describe('<Button /> ', () => {
    it('renders Button element', () => {
        const buttonName="click me"
        const RenderComponent = shallow(
            <Button buttonName={buttonName}></Button>
        );
        // console.log(RenderComponent)
        expect(RenderComponent.exists()).toBe(true)
    })

    it('recieved props correctly and button click', () => {
        const buttonName = "click me";
        // jest.fn()
        const mockFunction = sinon.spy()
        const RenderComponent = shallow(<Button buttonName={buttonName} onClicked={mockFunction}></Button>)
        RenderComponent.find('button').simulate('click')
        expect(mockFunction).toHaveProperty('callCount', 1)
        // console.log(RenderComponent)

    })


    it('Matches snapshot', () => {
        const buttonName = "click me";
        const render = shallow(<Button buttonName={buttonName}></Button>)
        expect(render).toMatchSnapshot()
    })
})