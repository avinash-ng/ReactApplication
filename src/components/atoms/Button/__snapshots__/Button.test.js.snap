// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button />  Matches snapshot 1`] = `ShallowWrapper {}`;
